pkg/controller/main-controller.go

```diff
@@ -428,8 +441,12 @@ func leaderRun(ctx context.Context, c *Controller, threadiness int, stopCh <-cha
 	// 2) we need to make sure we have STS API certificates (if enabled)
 	if IsSTSEnabled() {
 		go func() {
-			klog.Infof("STS is enabled, starting STS API certificate setup")
-			c.generateSTSTLSCert()
+			if utils.GetOperatorRuntime() == common.OperatorRuntimeOpenshift {
```

Comment:
Better we can create a boolean for this condition and use, as its used in multiple places. Something like
```
isOpenShift := utils.GetOperatorRuntime() == common.OperatorRuntimeOpenshift
```

Reply:
Can think of a global

Reply:
That's a temptation I had, but certainly Openshift is not the only one that will have a customized behavior, Rancher probably will have something similar or EKS, AKS,  etc. That is why I did not set it as a boolean.